Eatery/Controllers/BRB/BRBViewController.swift

```diff
@@ -15,70 +16,105 @@ class BRBViewController: UIViewController {
                                                             target: self,
                                                             action: #selector(aboutButtonPressed(_:)))
         navigationController?.navigationBar.prefersLargeTitles = true
         
-        connectionHandler = BRBConnectionHandler()
-        connectionHandler.delegate = self
-
+        accountManager = BRBAccountManager()
+        accountManager.delegate = self
+        loginViewController.accountManager = self.accountManager
+        
         loginViewController.delegate = self
         addChildViewController(loginViewController)
         view.addSubview(loginViewController.view)
         loginViewController.view.snp.makeConstraints { make in
             make.edges.equalToSuperview()
         }
         loginViewController.didMove(toParentViewController: self)
+        
+        if let account = accountManager.getCachedAccount() {
+            self.setState(.account(account))
+            loggedIn = true
+            if isLoading == nil {
+                isLoading = true
+            }
+        }
     }
     
     @objc private func aboutButtonPressed(_ sender: UIBarButtonItem) {
         let aboutVC = AboutTableViewController()
         aboutVC.delegate = self
         
-        if case .finished = connectionHandler.stage {
+        if case .finished = accountManager.getConnectionStage() {
+            aboutVC.logoutEnabled = true
+        } else if loggedIn {
             aboutVC.logoutEnabled = true
         } else {
             aboutVC.logoutEnabled = false
         }
         
         navigationController?.pushViewController(aboutVC, animated: true)
+        navigationItem.title = "Back"
     }
     
     private func setState(_ newState: State) {
         switch (state, newState) {
         case (.login, .account(let account)):
             let accountViewController = BRBAccountViewController(account: account)
             self.accountViewController = accountViewController
-            
             addChildViewController(accountViewController)
             view.insertSubview(accountViewController.view, at: 0)
             accountViewController.view.snp.makeConstraints { make in
                 make.edges.equalToSuperview()
             }
             accountViewController.didMove(toParentViewController: self)
             
+            activityIndicator = NVActivityIndicatorView(frame: CGRect(x: 0, y: 0, width: 22, height: 22), type: .circleStrokeSpin, color: .white)
+            let activityItem = UIBarButtonItem(customView: activityIndicator!)
+            navigationItem.setLeftBarButton(activityItem, animated: true)
+            
+            let refreshControl = UIRefreshControl(frame: .zero)
+            refreshControl.tintColor = .white
+            refreshControl.addTarget(self, action: #selector(refreshBRBAccount), for: .valueChanged)
+            self.accountViewController?.tableView.refreshControl = refreshControl
```

Comment:
I think the `accountViewController` should manage its own refresh control. Maybe have a property called `isRefreshing` in `accountViewController` that has a `didSet`. A `get set` property might work as well.

Reply:
Yeah I was thinking this too